test/suite/test_compact02.py

```diff
@@ -115,11 +115,9 @@ def test_compact02(self):
         c.close()
         self.pr('Removed total ' + str((count * 9666) / mb) + 'MB')
 
-        # 4. Reopen the connection to force the file to disk (if we leave dirty
-        # blocks in the cache, it can affect how compact works depending on how
-        # and when those blocks are written, because they're being written best
-        # fit, not first-fit, as compaction does).
-        self.reopen_conn()
+        # 4. Checkpoint to make the deleted records available for reuse, so
+        # compaction can be successful.
+        self.session.checkpoint()
```

Comment:
I agree, this probably isn't necessary, I'll pull it out, as well as the subsequent stat call.